packages/pigeon/lib/kotlin_generator.dart

```diff
@@ -0,0 +1,653 @@
+  indent.scoped('{', '}', () {
+    indent.write('companion object ');
+    indent.scoped('{', '}', () {
+      indent.writeln('/** The codec used by $apiName. */');
+      indent.write('val codec: MessageCodec<Any?> by lazy ');
+      indent.scoped('{', '}', () {
+        indent.writeln(_getCodecName(api));
+      });
+    });
+
+    for (final Method func in api.methods) {
+      final String channelName = makeChannelName(api, func);
+      final String returnType = func.returnType.isVoid
+          ? ''
+          : _nullsafeKotlinTypeForDartType(func.returnType);
+      String sendArgument;
+      if (func.arguments.isEmpty) {
+        indent.write('fun ${func.name}(callback: ($returnType) -> Unit) ');
+        sendArgument = 'null';
+      } else {
+        final Iterable<String> argTypes = func.arguments
+            .map((NamedType e) => _nullsafeKotlinTypeForDartType(e.type));
+        final Iterable<String> argNames =
+            indexMap(func.arguments, _getSafeArgumentName);
+        sendArgument = 'listOf(${argNames.join(', ')})';
+        final String argsSignature = map2(argTypes, argNames,
+            (String type, String name) => '$name: $type').join(', ');
+        if (func.returnType.isVoid) {
+          indent
+              .write('fun ${func.name}($argsSignature, callback: () -> Unit) ');
+        } else {
+          indent.write(
+              'fun ${func.name}($argsSignature, callback: ($returnType) -> Unit) ');
+        }
+      }
+      indent.scoped('{', '}', () {
+        const String channel = 'channel';
+        indent.writeln(
+            'val $channel = BasicMessageChannel<Any?>(binaryMessenger, "$channelName", codec)');
+        indent.write('$channel.send($sendArgument) ');
+        if (func.returnType.isVoid) {
+          indent.scoped('{', '}', () {
+            indent.writeln('callback()');
+          });
+        } else {
+          final String forceUnwrap = func.returnType.isNullable ? '?' : '';
+          indent.scoped('{', '}', () {
+            indent.writeln('val result = it as$forceUnwrap $returnType');
+            indent.writeln('callback(result)');
+          });
+        }
+      });
+    }
+  });
+}
+
+String _castForceUnwrap(String value, TypeDeclaration type, Root root) {
+  if (isEnum(root, type)) {
+    final String forceUnwrap = type.isNullable ? '' : '!!';
+    final String nullableConditionPrefix =
+        type.isNullable ? '$value == null ? null : ' : '';
+    return '$nullableConditionPrefix${_kotlinTypeForDartType(type)}.ofRaw($value as Int)$forceUnwrap';
+  } else {
+    final String castUnwrap = type.isNullable ? '?' : '';
+
+    // The StandardMessageCodec can give us [Integer, Long] for
+    // a Dart 'int'.  To keep things simple we just use 64bit
+    // longs in Pigeon with Kotlin.
+    if (type.baseName == 'int') {
+      return '$value.let { if (it is Int) it.toLong() else it as$castUnwrap Long }';
+    } else
```

Comment:
Nit: Missing braces (we always include them for the `else` if we have them on the `if`).

Reply:
Fixed